pandas/core/indexes/datetimelike.py

```diff
@@ -303,12 +304,18 @@ def map(self, mapper, na_action=None):
         except Exception:
             return self.astype(object).map(mapper)
 
-    def sort_values(self, return_indexer=False, ascending=True):
+    def sort_values(
+        self, return_indexer=False, ascending=True, key: Optional[Callable] = None
+    ):
         """
         Return sorted copy of Index.
         """
+        assert isinstance(self, Index)
```

Comment:
Is this required? Seems a little strange to add in a Mixin

Reply:
@WillAyd this is a mypy issue I haven't been able to figure out. It seems like type annotations on the mixin raise errors since it detects that `error: "DatetimeIndexOpsMixin" has no attribute "_get_attributes_dict"`. I'm not sure what to do about this except add the assert error.

Reply:
I've noticed that nothing in the `DatetimeIndexOpsMixin` is type annotated. Mypy does not seem to have a good system for handling them (see [here](https://stackoverflow.com/questions/51930339/how-do-i-correctly-add-type-hints-to-mixin-classes) or [here](https://github.com/python/mypy/issues/5837), for example). I would vote for not type annotating this function until we settle on a way to type annotate mixins generally.

Reply:
I've noticed that nothing in the `DatetimeIndexOpsMixin` is type annotated. Mypy does not seem to have a good system for handling them (see [here](https://stackoverflow.com/questions/51930339/how-do-i-correctly-add-type-hints-to-mixin-classes) or [here](https://github.com/python/mypy/issues/5837), for example). I would vote for not type annotating this function until we settle on a way to type annotate mixins generally.